boa_engine/src/symbol.rs

```diff
@@ -319,17 +209,82 @@ impl JsSymbol {
     /// [spec]: https://tc39.es/ecma262/#sec-symboldescriptivestring
     #[must_use]
     pub fn descriptive_string(&self) -> JsString {
-        self.inner.description.as_ref().map_or_else(
+        self.description().as_ref().map_or_else(
             || js_string!("Symbol()"),
             |desc| js_string!(utf16!("Symbol("), desc, utf16!(")")),
         )
     }
+
+    well_known_symbols! {
+        /// Gets the static `JsSymbol` for `"Symbol.asyncIterator"`.
+        (async_iterator, WellKnown::AsyncIterator),
+        /// Gets the static `JsSymbol` for `"Symbol.hasInstance"`.
+        (has_instance, WellKnown::HasInstance),
+        /// Gets the static `JsSymbol` for `"Symbol.isConcatSpreadable"`.
+        (is_concat_spreadable, WellKnown::IsConcatSpreadable),
+        /// Gets the static `JsSymbol` for `"Symbol.iterator"`.
+        (iterator, WellKnown::Iterator),
+        /// Gets the static `JsSymbol` for `"Symbol.match"`.
+        (r#match, WellKnown::Match),
+        /// Gets the static `JsSymbol` for `"Symbol.matchAll"`.
+        (match_all, WellKnown::MatchAll),
+        /// Gets the static `JsSymbol` for `"Symbol.replace"`.
+        (replace, WellKnown::Replace),
+        /// Gets the static `JsSymbol` for `"Symbol.search"`.
+        (search, WellKnown::Search),
+        /// Gets the static `JsSymbol` for `"Symbol.species"`.
+        (species, WellKnown::Species),
+        /// Gets the static `JsSymbol` for `"Symbol.split"`.
+        (split, WellKnown::Split),
+        /// Gets the static `JsSymbol` for `"Symbol.toPrimitive"`.
+        (to_primitive, WellKnown::ToPrimitive),
+        /// Gets the static `JsSymbol` for `"Symbol.toStringTag"`.
+        (to_string_tag, WellKnown::ToStringTag),
+        /// Gets the static `JsSymbol` for `"Symbol.unscopables"`.
+        (unscopables, WellKnown::Unscopables),
+    }
+}
+
+impl Clone for JsSymbol {
+    fn clone(&self) -> Self {
+        if let UnwrappedTagged::Ptr(ptr) = self.repr.unwrap() {
+            // SAFETY: the pointer returned by `self.repr` must be a valid pointer
+            // that came from an `Arc::into_raw` call.
+            unsafe {
+                let arc = Arc::from_raw(ptr.as_ptr().cast_const());
+                // Don't need the pointer since `self` is already a copyable pointer, just need to
+                // trigger the `clone` impl.
+                let _ = Arc::into_raw(arc.clone());
```

Comment:
How does this work exactly? Won't the reference count revert back as soon as this line executes as _ drops immediately?

Reply:
Ah, I think I over-engineered this a bit. Let me fix this.

Reply:
Changed it to explicitly use `mem::forget` instead.